woocommerce/woocommerce-ios
# [Composite Products] Display composite product and component hierarchy in order details

Closes: #8963
⚠️ UI changes depend on #9778


## Description

This PR adds support on the Networking layer to set the `parent` on an `OrderItem` when the item is a component in a composite product (with the Composite Products extension).

Along with the UI changes in #9778, this displays the parent/child hierarchy for a composite product and its child components in order details.

### Changes

* In `OrderItem`, sets the `parent` property as either the `bundled_by` field (for a bundled item) or the `composite_parent` field (for a component).
* Adds a mock response and test to confirm the `composite_parent` field is parsed correctly.

## Testing instructions


1. Create an order that includes a composite product.
2. In the app, open the Orders tab and select the order.
3. Confirm the composite product appears in the Products section with its components listed underneath and indented.

## Screenshots


Before|After
-|-
![composite-before](https://github.com/woocommerce/woocommerce-ios/assets/8658164/608570f2-e549-48b4-8a3a-2cc8c6ef782e)|![composite-after](https://github.com/woocommerce/woocommerce-ios/assets/8658164/2c348482-d823-4529-9d95-1ce2969d2de8)

---
